### PR TITLE
[23.2] fix s3fs templating

### LINKED ISSUE #17310: enable s3fs file_source plugin to be templated
**Describe the bug**

It seems that we are not able to define a generic_s3 endpoint that can be filled by user credentials.

```
- type: s3fs
  label: Generic S3 instance
  id: generic_s3
  doc: Access to a generic S3 endpoint.
  endpoint_url: ${user.preferences['generic_s3|endpoint_url']}
  bucket: ${user.preferences['generic_s3|bucket']}
  writable: true
  secret: ${user.preferences['generic_s3|secret']}
  key: ${user.preferences['generic_s3|key']}
```

It seems `_serialization_props` https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/files/sources/s3fs.py#L106 is not used.

### PR DESCRIPTION
**Pull request by bgruening**
this should fix https://github.com/galaxyproject/galaxy/issues/17310

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by bgruening** [comment]
I can list, and export to generic_s3 but not download. During download I get a "Parameter validation failed:
Invalid length for parameter Key, value: 0, valid min length: 1". Sp putting it into draft.

**Comment by bgruening** [comment]
@nuwang I rebased against 23.2 to also fix the export operations (export was not supported in 23.1). And I needed to remove `bucket` from the props in the ls() function.

Afaik, import/export/download is now all working with templates. It can be tested on EU.

File: lib/galaxy/files/sources/s3fs.py [hunk 1]
```diff
@@ -44,46 +44,60 @@ def __init__(self, **kwd: Unpack[S3FsFilesSourceProperties]):
...
+            bucket_path = self._bucket_path(_bucket_name, path)
             res = fs.ls(bucket_path, detail=True)
             to_dict = functools.partial(self._resource_info_to_dict, path)
             return list(map(to_dict, res))
 
     def _realize_to(self, source_path, native_path, user_context=None, opts: Optional[FilesSourceOptions] = None):
-        bucket_path = self._bucket_path(source_path)
-        self._open_fs(user_context=user_context, opts=opts).download(bucket_path, native_path)
+        _props = self._serialization_props(user_context)
+        _bucket_name = _props.get("bucket", "")
```
Comment (nuwang) [inline lib/galaxy/files/sources/s3fs.py]: I assume these should be popped too?
```suggestion
        _bucket_name = _props.pop("bucket", "")
```

File: lib/galaxy/files/sources/s3fs.py [hunk 2]
```diff
@@ -44,46 +44,60 @@ def __init__(self, **kwd: Unpack[S3FsFilesSourceProperties]):
...
+        fs = self._open_fs(props=_props, opts=opts)
+        bucket_path = self._bucket_path(_bucket_name, source_path)
+        fs.download(bucket_path, native_path)
 
     def _write_from(self, target_path, native_path, user_context=None, opts: Optional[FilesSourceOptions] = None):
-        bucket_path = self._bucket_path(target_path)
-        self._open_fs(user_context=user_context, opts=opts).upload(native_path, bucket_path)
+        _props = self._serialization_props(user_context)
+        _bucket_name = _props.get("bucket", "")
```
Comment (nuwang) [inline lib/galaxy/files/sources/s3fs.py]:
```suggestion
        _bucket_name = _props.pop("bucket", "")
```

File: lib/galaxy/files/sources/s3fs.py (hunk 1 above)
Comment (bgruening) [reply lib/galaxy/files/sources/s3fs.py]: get works here and below

Comment (nuwang) [reply lib/galaxy/files/sources/s3fs.py]: Test is failing? https://github.com/galaxyproject/galaxy/actions/runs/7568788545/job/20610690860?pr=17311#step:8:3114

**Review by nuwang — APPROVED** [review]
lgtm!

**Comment by bgruening** [comment]
Cool, thanks a lot @nuwang for your help!

**Pull request merged**